.gitignore

```diff
@@ -11,3 +11,4 @@
 /test_scripts/test.cfg
```

Comment:
The vast majority of the changes are generated by perltidy, which is a popular and very commonly-used code formatter:

https://metacpan.org/pod/distribution/Perl-Tidy/bin/perltidy

This is the first in a series of updates to the perl templates that I have made and have been using on my own repo. I have added in tests for perl file compilation and rewritten the test script to use the perl test harness properly (I have also retrofitted those changes to a couple of the existing perl kbase apps -- e.g. RAST SDK; this means test failures will be reported, rather than just dying silently). This PR just applies some standardised formatting.